relay-metrics/src/aggregator.rs

```diff
@@ -742,34 +747,48 @@ impl Aggregator {
             self.config.max_project_key_bucket_bytes,
         )?;
 
-        let added_cost;
-        match self.buckets.entry(key) {
-            Entry::Occupied(mut entry) => {
-                relay_statsd::metric!(
-                    counter(MetricCounters::MergeHit) += 1,
-                    aggregator = &self.name,
-                    namespace = entry.key().namespace().as_str(),
-                );
+        let mut added_cost = 0;
 
-                added_cost = entry.get_mut().merge(bucket.value, bucket.metadata)?;
-            }
-            Entry::Vacant(entry) => {
-                relay_statsd::metric!(
-                    counter(MetricCounters::MergeMiss) += 1,
-                    aggregator = &self.name,
-                    namespace = entry.key().namespace().as_str(),
-                );
-                relay_statsd::metric!(
-                    set(MetricSets::UniqueBucketsCreated) = entry.key().hash64() as i64, // 2-complement
-                    aggregator = &self.name,
-                    namespace = entry.key().namespace().as_str(),
-                );
+        let mut error = None;
+        let updated = self.buckets.change_priority_by(&key, |value| {
```

Comment:
Why do we not use `self.buckets.get_mut()` here? IIUC, the closure is only called if the element exists, and we don't actually update any priorities.

Reply:
`get_mut` only returns the key as mutable, but we need a mutable value which also causes this unfortunate error handling.